packages/model/addon/-private/system/many-array.js

```diff
@@ -140,39 +136,33 @@ export default EmberObject.extend(MutableArray, DeprecatedEvented, {
     @private
     */
     this.currentState = [];
-    this.flushCanonical(this.initialState, false);
-    // we don't need this anymore, it just prevents garbage collection the records in the initialState
-    this.initialState = undefined;
+    this._isUpdating = false;
+    this._isDirty = false;
+    // make sure we initialize to the correct state
+    // since the user has already accessed
+    this.retrieveLatest();
   },
 
-  // TODO: if(DEBUG)
-  anyUnloaded() {
-    // Use `filter[0]` as opposed to `find` because of IE11
-    let unloaded = this.currentState.filter((im) => im._isDematerializing || !im.currentState.isLoaded)[0];
-    return !!unloaded;
+  // TODO refactor away _hasArrayObservers for tests
+  get _hasArrayObservers() {
+    return this.hasArrayObservers || this.__hasArrayObservers;
```

Comment:
Why do we have both of these?

Reply:
The first is a built in from Ember, the second is a prop for us to override for tests where we want to trigger the observer codepath but didn't necessarily add an observer (or do, but need to control the timing of when the ManyArray believes it has an observer)